src/flyte/remote/_client/auth/_keyring.py

```diff
@@ -108,10 +109,13 @@ def retrieve(for_endpoint: str) -> typing.Optional[Credentials]:
             logger.debug(f"Failed to retrieve tokens from keyring. Error: {e}")
             return None
 
-        if not access_token:
+        if not access_token and not refresh_token:
             logger.debug("No access token found in keyring.")
             return None
 
+        if not access_token and refresh_token:
+            access_token = ""
```

Comment:
confused by this... maybe it's the jet lag, wouldn't access_token already be 'not'?